### ISSUE TITLE
Linking error with v2.4.0 and Qt6

### ISSUE THREAD

**Issue by petrmanek** [cneben/QuickQanava#220]
Hi! I am getting the following linking error under the latest Qt6 and v2.4.0:

```
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE]+0x10): undefined reference to `qcm::AbstractContainer::metaObject() const'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE]+0x18): undefined reference to `qcm::AbstractContainer::qt_metacast(char const*)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4NodeEEE]+0x20): undefined reference to `qcm::AbstractContainer::qt_metacall(QMetaObject::Call, int, void**)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTIN3qcm9ContainerI7QVectorPN3qan4NodeEEE[_ZTIN3qcm9ContainerI7QVectorPN3qan4NodeEEE]+0x10): undefined reference to `typeinfo for qcm::AbstractContainer'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE]+0x10): undefined reference to `qcm::ContainerModel::metaObject() const'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE]+0x18): undefined reference to `qcm::ContainerModel::qt_metacast(char const*)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE]+0x20): undefined reference to `qcm::ContainerModel::qt_metacall(QMetaObject::Call, int, void**)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTIN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE[_ZTIN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4NodeEEES5_EE]+0x10): undefined reference to `typeinfo for qcm::ContainerModel'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE]+0x10): undefined reference to `qcm::AbstractContainer::metaObject() const'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE]+0x18): undefined reference to `qcm::AbstractContainer::qt_metacast(char const*)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan4EdgeEEE]+0x20): undefined reference to `qcm::AbstractContainer::qt_metacall(QMetaObject::Call, int, void**)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTIN3qcm9ContainerI7QVectorPN3qan4EdgeEEE[_ZTIN3qcm9ContainerI7QVectorPN3qan4EdgeEEE]+0x10): undefined reference to `typeinfo for qcm::AbstractContainer'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE]+0x10): undefined reference to `qcm::ContainerModel::metaObject() const'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE]+0x18): undefined reference to `qcm::ContainerModel::qt_metacast(char const*)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE[_ZTVN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE]+0x20): undefined reference to `qcm::ContainerModel::qt_metacall(QMetaObject::Call, int, void**)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTIN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE[_ZTIN3qcm18ContainerModelImplINS_9ContainerI7QVectorPN3qan4EdgeEEES5_EE]+0x10): undefined reference to `typeinfo for qcm::ContainerModel'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE]+0x10): undefined reference to `qcm::AbstractContainer::metaObject() const'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE]+0x18): undefined reference to `qcm::AbstractContainer::qt_metacast(char const*)'
/usr/bin/ld: lib/quickqanava/src/libQuickQanava.a(qanGraph.cpp.o):(.data.rel.ro._ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE[_ZTVN3qcm9ContainerI7QVectorPN3qan5GroupEEE]+0x20): undefined reference to `qcm::AbstractContainer::qt_metacall(QMetaObject::Call, int, void**)'
```

I suppose that QuickContainers were omitted from library sources, applying the following patch to harmonize `src/CMakeLists.qt6.txt` with `src/CMakeLists.qt5.txt` fixes the problem for me:

```diff
diff --git a/src/CMakeLists.qt6.txt b/src/CMakeLists.qt6.txt
index 0da14de..d15d501 100644
--- a/src/CMakeLists.qt6.txt
+++ b/src/CMakeLists.qt6.txt
@@ -14,7 +14,7 @@ set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS $<$<CONFIG:Debug>:QT_
 # Configure QuickQanava library ###############################################
 list(APPEND qan_resources QuickQanava_static.qrc)
 list(APPEND qan_resources GraphicalEffects6/QuickQanavaGraphicalEffects.qrc)
-add_library(QuickQanava STATIC ${qan_source_files} ${qan_header_files} ${qan_resources})
+add_library(QuickQanava STATIC ${qan_source_files} ${qan_header_files} ${qan_resources} ${quickcontainers_source_files} ${quickcontainers_header_files})
 
 target_include_directories(QuickQanava
     PUBLIC
```

**Comment by cneben** [comment]
Should be fixed in 2.4.1 maintenance release (https://github.com/cneben/QuickQanava/releases/tag/2.4.1qt5) can I let you close @petrmanek please ?

**Comment by petrmanek** [comment]
> Should be fixed in 2.4.1 maintenance release (https://github.com/cneben/QuickQanava/releases/tag/2.4.1qt5) can I let you close @petrmanek please ?

I can confirm that 2.4.1 works nicely with Qt6. Closing as completed.
